[Bug]: Can not set configuration when setting UGI

### What happened?

When using Flink to read the existing table, A Kerberos authentication error occurred.

### Affects Versions

master

### What engines are you seeing the problem on?

Flink

### How to reproduce

Use the Amoro Flink runtime jar in Flink to create an Amoro table and execute "select * from xxx" SQL.

### Relevant log output

```shell
Caused by: org.apache.flink.table.catalog.exceptions.CatalogException: transformer from meta failed.
	at com.netease.sloth.flink.connector.arctic.catalog.ArcticCatalogWrapper.getTable(ArcticCatalogWrapper.java:130)
	at org.apache.flink.table.catalog.CatalogManager.getPermanentTable(CatalogManager.java:425)
	at org.apache.flink.table.catalog.CatalogManager.getTable(CatalogManager.java:395)
	at org.apache.flink.table.planner.delegation.PlannerBase.getTableSink(PlannerBase.scala:356)
	at org.apache.flink.table.planner.delegation.PlannerBase.translateToRel(PlannerBase.scala:222)
	at org.apache.flink.table.planner.delegation.PlannerBase.$anonfun$translate$1(PlannerBase.scala:182)
	at scala.collection.TraversableLike.$anonfun$map$1(TraversableLike.scala:237)
	at scala.collection.Iterator.foreach(Iterator.scala:941)
	at scala.collection.Iterator.foreach$(Iterator.scala:941)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
	at scala.collection.IterableLike.foreach(IterableLike.scala:74)
	at scala.collection.IterableLike.foreach$(IterableLike.scala:73)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:56)
	at scala.collection.TraversableLike.map(TraversableLike.scala:237)
	at scala.collection.TraversableLike.map$(TraversableLike.scala:230)
	at scala.collection.AbstractTraversable.map(Traversable.scala:108)
	at org.apache.flink.table.planner.delegation.PlannerBase.translate(PlannerBase.scala:182)
	at com.netease.sloth.flink114.sql.table.FlinkTableEnvironmentImpl.translate(FlinkTableEnvironmentImpl.java:533)
	at com.netease.sloth.flink114.sql.table.FlinkTableEnvironmentImpl.sqlUpdateWithoutUdfClassloader(FlinkTableEnvironmentImpl.java:289)
	at com.netease.sloth.flink114.sql.table.FlinkTableEnvironmentImpl.lambda$sqlUpdate$2(FlinkTableEnvironmentImpl.java:239)
	at com.netease.sloth.flink.sql.api.classload.ClassLoadWrap.run(ClassLoadWrap.java:88)
	... 38 more
Caused by: java.lang.IllegalArgumentException: Can't get Kerberos realm
	at org.apache.hadoop.security.HadoopKerberosName.setConfiguration(HadoopKerberosName.java:65)
	at org.apache.hadoop.security.UserGroupInformation.initialize(UserGroupInformation.java:276)
	at org.apache.hadoop.security.UserGroupInformation.setConfiguration(UserGroupInformation.java:312)
	at com.netease.arctic.table.TableMetaStore.constructUgi(TableMetaStore.java:333)
	at com.netease.arctic.table.TableMetaStore.getUGI(TableMetaStore.java:228)
	at com.netease.arctic.table.TableMetaStore.doAs(TableMetaStore.java:343)
	at com.netease.arctic.io.ArcticHadoopFileIO.exists(ArcticHadoopFileIO.java:192)
	at com.netease.arctic.hive.catalog.MixedHiveTables.checkPrivilege(MixedHiveTables.java:84)
	at com.netease.arctic.hive.catalog.MixedHiveTables.loadKeyedTable(MixedHiveTables.java:65)
	at com.netease.arctic.hive.catalog.MixedHiveTables.loadKeyedTable(MixedHiveTables.java:38)
	at com.netease.arctic.catalog.MixedTables.loadTableByMeta(MixedTables.java:79)
	at com.netease.arctic.catalog.BasicArcticCatalog.loadTable(BasicArcticCatalog.java:154)
	at com.netease.arctic.catalog.ArcticCatalog.tableExists(ArcticCatalog.java:105)
	at com.netease.arctic.flink.catalog.ArcticCatalog.getTable(ArcticCatalog.java:173)
	at com.netease.sloth.flink.connector.arctic.catalog.ArcticCatalogWrapper.getTable(ArcticCatalogWrapper.java:124)
	... 58 more
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.hadoop.security.authentication.util.KerberosUtil.getDefaultRealm(KerberosUtil.java:84)
	at org.apache.hadoop.security.HadoopKerberosName.setConfiguration(HadoopKerberosName.java:63)
	... 72 more
Caused by: KrbException: Cannot locate default realm
	at sun.security.krb5.Config.getDefaultRealm(Config.java:1137)
	... 78 more
```


### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
+++ b/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
@@ -426,7 +426,12 @@ public class TableMetaStore implements Serializable {
     return String.format("%s/%s", confPath.toString(), confName);
   }
 
-  private static Configuration buildConfiguration(TableMetaStore metaStore) {
+  private Configuration buildConfiguration(TableMetaStore metaStore) {
+    if (TableMetaStore.AUTH_METHOD_KERBEROS.equals(metaStore.authMethod)) {
+      generateKrbConfPath();
+      String krbConfFile = saveConfInPath(confCachePath, KRB_CONF_FILE_NAME, krbConf);
+      System.setProperty(KRB5_CONF_PROPERTY, krbConfFile);
+    }
     Configuration configuration = new Configuration();
     configuration.addResource(new ByteArrayInputStream(metaStore.getCoreSite()));
     configuration.addResource(new ByteArrayInputStream(metaStore.getHdfsSite()));

--- a/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
+++ b/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
@@ -35,6 +35,7 @@ import org.apache.iceberg.relocated.com.google.common.hash.Hashing;
 import org.apache.iceberg.relocated.com.google.common.io.ByteStreams;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import sun.security.krb5.KrbException;
 
 import java.io.ByteArrayInputStream;
 import java.io.FileInputStream;
@@ -228,7 +229,7 @@ public class TableMetaStore implements Serializable {
           constructUgi();
         }
         LOG.info("Complete to build ugi {}", authInformation());
-      } catch (IOException e) {
+      } catch (IOException | KrbException e) {
         throw new RuntimeException("Fail to init user group information", e);
       }
     } else {
@@ -325,11 +326,12 @@ public class TableMetaStore implements Serializable {
     return authInformation;
   }
 
-  private void constructUgi() throws IOException {
+  private void constructUgi() throws IOException, KrbException {
     String krbConfFile = saveConfInPath(confCachePath, KRB_CONF_FILE_NAME, krbConf);
     String keyTabFile = saveConfInPath(confCachePath, KEY_TAB_FILE_NAME, krbKeyTab);
     System.clearProperty(HADOOP_USER_PROPERTY);
     System.setProperty(KRB5_CONF_PROPERTY, krbConfFile);
+    sun.security.krb5.Config.refresh();
     UserGroupInformation.setConfiguration(getConfiguration());
     KerberosName.resetDefaultRealm();
     ugi = UserGroupInformation.loginUserFromKeytabAndReturnUGI(krbPrincipal, keyTabFile);
@@ -427,11 +429,6 @@ public class TableMetaStore implements Serializable {
   }
 
   private Configuration buildConfiguration(TableMetaStore metaStore) {
-    if (TableMetaStore.AUTH_METHOD_KERBEROS.equals(metaStore.authMethod)) {
-      generateKrbConfPath();
-      String krbConfFile = saveConfInPath(confCachePath, KRB_CONF_FILE_NAME, krbConf);
-      System.setProperty(KRB5_CONF_PROPERTY, krbConfFile);
-    }
     Configuration configuration = new Configuration();
     configuration.addResource(new ByteArrayInputStream(metaStore.getCoreSite()));
     configuration.addResource(new ByteArrayInputStream(metaStore.getHdfsSite()));

--- a/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
+++ b/core/src/main/java/com/netease/arctic/table/TableMetaStore.java
@@ -428,7 +428,7 @@ public class TableMetaStore implements Serializable {
     return String.format("%s/%s", confPath.toString(), confName);
   }
 
-  private Configuration buildConfiguration(TableMetaStore metaStore) {
+  private static Configuration buildConfiguration(TableMetaStore metaStore) {
     Configuration configuration = new Configuration();
     configuration.addResource(new ByteArrayInputStream(metaStore.getCoreSite()));
     configuration.addResource(new ByteArrayInputStream(metaStore.getHdfsSite()));

--- a/docs/engines/trino.md
+++ b/docs/engines/trino.md
@@ -25,6 +25,10 @@ please refer to the documentation at [Iceberg Connector](https://trino.io/docs/c
 connector.name=arctic
 arctic.url=thrift://{ip}:{port}/{catalogName}
 ```
+- Configure the JVM configuration file for Trino in the {trino_home}/etc directory named `jvm.config` :
+```tex
+--add-exports=java.security.jgss/sun.security.krb5=ALL-UNNAMED
+```
 
 ### Support SQL statement
 

--- a/docs/engines/trino.md
+++ b/docs/engines/trino.md
@@ -20,7 +20,6 @@ please refer to the documentation at [Iceberg Connector](https://trino.io/docs/c
 - Create the {trino_home}/plugin/amoro directory in the Trino installation package, 
   and extract the contents of the amoro-trino package trino-amoro-xx-SNAPSHOT.tar.gz to the {trino_home}/plugin/amoro directory.
 - Configure the Catalog configuration file for Amoro in the {trino_home}/etc/catalog directory, for example:
-- 
 ```tex
 connector.name=arctic
 arctic.url=thrift://{ip}:{port}/{catalogName}

